src/test/java/com/team4/testingsystem/services/impl/FileAnswerServiceImplTest.java

```diff
@@ -0,0 +1,102 @@
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Optional;
+
+@ExtendWith(MockitoExtension.class)
+class FileAnswerServiceImplTest {
+
+    @Mock
+    private QuestionRepository questionRepository;
+
+    @Mock
+    private FileAnswerRepository fileAnswerRepository;
+
+    @Mock
+    private FileAnswer fileAnswer;
+
+    @Mock
+    private Question question;
+
+    @InjectMocks
+    private FileAnswerServiceImpl fileAnswerService;
+
+    @Test
+    void getByIdSuccess() {
+        Mockito.when(fileAnswerRepository.findById(fileAnswer.getId())).thenReturn(Optional.of(fileAnswer));
+        Assertions.assertEquals(fileAnswer, fileAnswerService.getById(fileAnswer.getId()));
+    }
+
+    @Test
+    void getByIdFail() {
+        Mockito.when(fileAnswerRepository.findById(5L)).thenThrow(new NotFoundException());
+        Assertions.assertThrows(NotFoundException.class, () -> fileAnswerService.getById(5L));
+
+    }
+
+    @Test
+    void createSuccess() {
+        FileAnswerRequest fileAnswerRequest = FileAnswerRequest.builder().url("").questionId(1L).build();
+        Mockito.when(questionRepository.findById(1L)).thenReturn(Optional.of(question));
+        Assertions.assertDoesNotThrow(() -> fileAnswerService.create(fileAnswerRequest));
+    }
+
+    @Test
+    void createFail() {
+        FileAnswerRequest fileAnswerRequest = FileAnswerRequest.builder().url("").questionId(1L).build();
+        Mockito.when(questionRepository.findById(1L)).thenThrow(NotFoundException.class);
+        Assertions.assertThrows(NotFoundException.class, () -> fileAnswerService.create(fileAnswerRequest));
+    }
+
+    @Test
+    void updateSuccess() {
+        FileAnswerRequest fileAnswerRequest = FileAnswerRequest.builder().url("").questionId(10L).build();
+        Mockito.when(fileAnswerRepository.findById(1L)).thenReturn(Optional.of(fileAnswer));
+        Mockito.when(questionRepository.findById(10L)).thenReturn(Optional.of(question));
+        Assertions.assertDoesNotThrow(() -> fileAnswerService.update(1L, fileAnswerRequest));
+        Mockito.verify(fileAnswerRepository).save(Mockito.any());
+    }
+
+    @Test
+    void updateNotFoundInFileAnswerRepository() {
+        FileAnswerRequest fileAnswerRequest = FileAnswerRequest.builder().url("").questionId(10L).build();
+        Mockito.when(fileAnswerRepository.findById(1L)).thenThrow(NotFoundException.class);
+        Assertions.assertThrows(NotFoundException.class, () -> fileAnswerService.update(1L, fileAnswerRequest));
+        Mockito.verify(fileAnswerRepository, Mockito.times(0)).save(Mockito.any());
+    }
+
+    @Test
+    void updateNotFoundInQuestionRepository() {
+        FileAnswerRequest fileAnswerRequest = FileAnswerRequest.builder().url("").questionId(10L).build();
```

Comment:
Is it really more convenient, than just `new FileAnswerRequest("", 10L);`?)

Reply:
Just a common way. Sure, it's quite a small entity. For the bigger classes, it could be a bit more difficult. And they will look like new ...("...", "...", "...", "..."). It's too easy to make a mistake with an order here. Using builder make everything clearly